app/controllers/rentals_controller.rb

```diff
@@ -23,20 +34,42 @@ def index
   # GET /rentals/1
   def show; end
 
-  # GET /rentals/cost?end_time=time&start_time=time&item_type=1
+  # GET /rentals/cost?end_time=time&start_time=time&item_types=...
   def cost
-    start_time = Time.zone.parse(params[:start_time]).to_date.to_s
-    end_time = Time.zone.parse(params[:end_time]).to_date.to_s
-    item_type = ItemType.find(params[:item_type]) if params[:item_type]
-    render json: Rental.cost(start_time, end_time, item_type)
+    _params = cost_params
+    required_params = %w(start_time end_time item_types)
+    if (_params.to_h.keys & required_params) == required_params
+      start_time = Time.zone.parse(_params[:start_time]).to_date.to_s
+      end_time = Time.zone.parse(_params[:end_time]).to_date.to_s
+
+      begin
+        cost = _params[:item_types].each_with_object({}) do |it_id, acc|
+          if it = ItemType.find_by(id: it_id)
+            acc[it.name] ||= 0
+            acc[it.name] += it.cost(start_time, end_time)
+          else
+            raise ArgumentError, it_id
+          end
+        end
+      rescue => err
+        render(json: { errors: ["item not found #{err.message}"] }, status: 400) && (return)
```

Comment:
When you do "and return" you actually want the word "and"

Reply:
Yes, and also, there's no reason to write parentheses around a single word.

Reply:
in this case isn't `&&` is equivalent to `and`?

Reply:
I mean yes, in this case, because you wrapped everything in parentheses. But they're for different things. The logical `&&` is for, well, logic — like in a conditional. `and` is for control flow.